client/bootloader_flash.py

```diff
@@ -52,7 +52,7 @@ def flash_binary(fdesc, binary, base_address, device_class, destinations, page_s
     # First erase all pages
     for offset in range(0, len(binary), page_size):
         erase_command = commands.encode_erase_flash_page(base_address + offset, device_class)
-        utils.write_command(fdesc, erase_command, destinations)
+        utils.write_command_retry(fdesc, erase_command, destinations)
```

Comment:
The actual return value is not checked and the destinations list shouldn't contain any devices which failed.
The bootloader can also return a false eg. for a wrong device class in the flash write command. This case should be treated differently than a timeout. (This should probably be specified more clearly in the Readme)